app/redux/actions/async.js

```diff
@@ -940,25 +942,57 @@ export function fetchPatientData(api, options, id) {
       // passing in an initial startDate param in the patientdata.js initial data fetcher.
       if (options.initialStartDate) initialFetchParams.startDate = options.initialStartDate;
 
-      api.patientData.get(id, initialFetchParams, (err, results) => {
-        if (err) {
-          dispatch(sync.fetchPatientDataFailure(
-            createActionError(ErrorMessages.ERR_FETCHING_PATIENT_DATA, err), err
-          ));
-        } else {
+      const initialFetchers = {
+        serverTime: api.server.getTime.bind(api),
+        latestDatums: api.patientData.get.bind(api, id, initialFetchParams),
+      };
+
+      async.parallel(async.reflectAll(initialFetchers), (err, results) => {
+        const resultsErr = _.mapValues(results, ({error}) => error);
+        const resultsVal = _.mapValues(results, ({value}) => value);
+        const hasError = _.some(resultsErr, err => !_.isUndefined(err));
+
+        if (hasError) {
+          handleFetchErrors(resultsErr);
+        }
+        else {
+          // On the initial fetch, we want to use the server time if we can in case the user's local
+          // computer time is off and set the max endDate to one day in the future since we can get
+          // `time` fields that are slightly (or not-so-slightly) in the future due to incorrect
+          // device and/or computer time upon upload.
+          const serverTime = _.get(resultsVal.serverTime, 'data.time');
+          dispatch(sync.fetchServerTimeSuccess(serverTime));
+
           // We determine the date range to fetch data for by first finding the latest
           // diabetes datum time and going back 30 days
-          const diabetesDatums = _.reject(results, d => _.includes(['food', 'upload'], d.type));
-          const latestDatumTime = _.max(_.map(diabetesDatums, d => (d.time)));
-          options.startDate = moment.utc(latestDatumTime || options.browserTimeStub).subtract(30, 'days').startOf('day').toISOString();
-          options.endDate = moment.utc(latestDatumTime || options.browserTimeStub).add(1, 'days').toISOString();
+          const diabetesDatums = _.reject(resultsVal.latestDatums, d => _.includes(['food', 'upload'], d.type));
+          const latestDiabetesDatumTime = _.max(_.map(diabetesDatums, d => (d.time)));
+
+          if (moment.utc(latestDiabetesDatumTime).diff(moment.utc(serverTime), 'days') > 1) {
+            _.isFunction(rollbar.error) && rollbar.error(
```

Comment:
Should we consider using `rollbar.info` (a la https://github.com/tidepool-org/uploader/blob/2752f10c2a4f741785d5811c9cf3929a7fa176d0/lib/commonFunctions.js#L144 ) instead of treating this as an `Error`, since it's largely informational (no specific action for us to take/fix)

Reply:
Good idea.  I'll change it.